packages/expo-app-loading/src/AppLoading.tsx

```diff
@@ -0,0 +1,101 @@
+import { EventEmitter } from 'fbemitter';
+import React from 'react';
+
+import NativeAppLoading from './AppLoadingNativeWrapper';
+
+type Props = {
+  /**
+   * Optional, you can do this process manually if you prefer.
+   * This is mainly for backwards compatibility and it is not recommended.
+   *
+   * When provided, requires providing `onError` prop as well.
+   * @deprecated
+   */
+  startAsync: () => Promise<void>;
+
+  /**
+   * If `startAsync` throws an error, it is caught and passed into the provided function.
+   * @deprecated
+   */
+  onError: (error: Error) => void;
+
+  /**
+   * Called when `startAsync` resolves or rejects.
+   * This should be used to set state and unmount the `AppLoading` component.
+   * @deprecated
+   */
+  onFinish?: () => void;
```

Comment:
`startAsync`, `onError`, and `onFinish` are all meant to be optional because you could just mount the `AppLoading` component without any props (this is how we used to recommend using it) and do your own work in `useEffect` or something.